wandb/sdk/wandb_run.py

```diff
@@ -182,6 +183,34 @@ def join(self) -> None:
         self._retry_thread.join()
 
 
+def attach(func: Callable) -> Callable:
+    @functools.wraps(func)
+    def wrapper(self: Type["Run"], *args: Any, **kwargs: Any) -> Any:
+
+        if self._is_attaching:
+            message = f"Trying to attach `{func.__name__}` while in the middle of attaching `{self._is_attaching}`"
+            raise RuntimeError(message)
```

Comment:
should it be an assertion error?

Reply:
i have no pref, seems reasonable as this is a base exception.